rpc/types/src/json.rs

```diff
@@ -49,8 +49,33 @@ define_request_and_response! {
         // $FIELD_NAME: $FIELD_TYPE,
         // ```
         // The struct generated and all fields are `pub`.
-        extra_nonce: String,
-        prev_block: String,
+
+        // This optional expression can be placed after
+        // a `field: field_type`. this indicates to the
+        // macro to (de)serialize this field using this
+        // default expression if it doesn't exist in epee.
+        //
+        // See `cuprate_epee_encoding::epee_object` for info.
+        //
+        // The default function must be specified twice:
+        //
+        // 1. As an expression
+        // 2. As a string literal
+        //
+        // For example: `extra_nonce: String /* = default_string(), "default_string" */,`
+        //
+        // This is a HACK since `serde`'s default attribute only takes in
+        // string literals and macros (stringify) within attributes do not work.
+        extra_nonce: String /* = default_expression, "default_literal" */,
```

Comment:
The type generator macro now handles `serde` default values directly.

Not ideal: attributes take in a string literal, so we can't pass `stringify!($default)` to `serde`'s `default(...)`.

So we have to provide:
- An expression for `cuprate_epee_encoding`
- A string literal for `serde`

Reply:
hmm that is annoying, I can't think of a way to work around this atm though